workshops/memory game/README.md

```diff
@@ -0,0 +1,555 @@
+- [setAttribute()](https://developer.mozilla.org/en-US/docs/Web/API/Element/setAttribute) sets the value of an attribute on the specified element 
+- [data attribute](https://developer.mozilla.org/en-US/docs/Learn/HTML/Howto/Use_data_attributes) allow us to store extra information on standard in the HTML tag. 
+
+According to the game, we have to flip the card that is clicked. So add a `click` event-listener for the card. Every time a card get clicked, `flipCard` function will be executed, which we will define later on the flow. For now, comment the event-listener as `flipCard` haven't yet defined.
+
+```js
+card.addEventListener('click', flipCard)
+```
+
+Then add the `card` into the `board` using `appendChild()`. 
+
+```js
+board.appendChild(card)
+```
+
+The method [appenChild()](https://developer.mozilla.org/en-US/docs/Web/API/Node/appendChild) adds a node to the end of the list of children of a specified parent node.
+
+The function we created finally, will be :
+
+```javascript
+const board = document.querySelector('.board')
+const placeholder = 'https://cloud-5ystxzer7.vercel.app/7placeholder.png'
+
+// create game board
+function createBoard() {
+  for (let i = 0; i < cardArray.length; i++) {
+    var card = document.createElement('img')
+    card.setAttribute('src', placeholder)
+    card.setAttribute('data-id', i)
+    //card.addEventListener('click', flipCard)
+    board.appendChild(card)
+  }
+}
+createBoard()
+```
+
+Press the run button and see the your game board. 
+
+It looks like this with all the cards turned down (i.e., placeholder images).
+
+![Temporary output](https://cloud-2vppbtbp1.vercel.app/0jsop.png)
+
+## Flip Card
+
+When a card is clicked, let's flip the card. 
+
+Create two empty variable arrays `cardsClicked` and `cardsClickedId`. Also create a variable `cardsMatched` array to push the matched cards
+
+```js
+var cardsClicked = []
+var cardsClickedId = []
+var cardsMatched = []
+```
+
+Create a function `flipCard()`. Then inside this function, create a variable `cardId`, which is the `data-id` attribute of the clicked card, using `getAttribute()`. 
+
+```js
+function flipCard() {
+  var cardId = this.getAttribute('data-id')
+  // code goes here
+}
+```
+
+Now add name of this card into `cardsClicked` array based on `cardId` using `push()` method. Also push the id of this card (i.e., `cardId`) into `cardsClickedId` array. 
+
+```js
+cardsClicked.push(cardArray[cardId].name)
+cardsClickedId.push(cardId)
+```
+
+Then add the front side of the card, image in `cardArray` corrresponding to `cardId`, using `setAttribute`.
+
+```js
+this.setAttribute('src', cardArray[cardId].img)
+```
+
+After selecting two cards, we have to check for match. For that, if two cards are clicked i.e., the length of `cardsClicked` array is equal to `2`, invoke `checkForMatch()` inside `setTimeout()` so that everything doesn't happen too fast.
+
+```js
+if (cardsClicked.length ===2) {
```

Comment:
weird spacing, should either be `cardsClicked.length===2` or `cardsClicked.length === 2`